### PR TITLE
feat(addon-doc): new token `TUI_DOC_PAGE_LOADED` to fix anchor links (scroll to examples)

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [X] Documentation content changes

## What is the current behavior?
* Try go to https://taiga-ui.dev/components/input#table
* It should scroll to example "Table" 
* **But**:

https://user-images.githubusercontent.com/35179038/176679073-aa4593f1-a902-4bc7-a202-c00dd7e49f4c.mov

## What is the new behavior?

https://user-images.githubusercontent.com/35179038/176679218-3d12725c-b9a7-4cf8-b53c-478c615ef4ad.mov

**You can try by yourself:**
https://taiga-ui--pr2031-fix-scroll-to-exampl-50r2a996.web.app/components/input#table

**Screenshot diff:**

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/demo/src/modules/app/utils/ready-to-scroll-factory.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {ElementRef} from '@angular/core';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+export function readyToScrollFactory(
```
Comment (splincode) [inline projects/demo/src/modules/app/utils/ready-to-scroll-factory.ts:5]: `tuiReadyToScrollFactory`

Comment (splincode) [reply projects/demo/src/modules/app/utils/ready-to-scroll-factory.ts:5]: oh, it's demo

File: projects/addon-doc/src/tokens/page-loaded.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {InjectionToken} from '@angular/core';
+import {defer, Observable, of, timer} from 'rxjs';
+import {switchMapTo} from 'rxjs/operators';
+
+export const TUI_DOC_PAGE_LOADED = new InjectionToken<Observable<boolean>>(
+    'Stream that emits if loading of page is over (for example, to begin scrollIntoView)',
```
Comment (splincode) [inline projects/addon-doc/src/tokens/page-loaded.ts:6]: `[TUI_DOC_PAGE_LOADED] Stream that emits...`

File: projects/addon-doc/src/components/navigation/navigation.component.ts [hunk 3]
```diff
@@ -163,9 +165,11 @@ export class TuiDocNavigationComponent {
     }
 
     private handleAnchorLink(hash: string): void {
-        setTimeout(() => {
-            this.navigateToAnchorLink(hash);
-        }, SCROLL_INTO_VIEW_DELAY);
+        this.readyToScroll$
+            .pipe(filter(Boolean), take(1), takeUntil(this.destroy$))
+            .subscribe(() => {
```
Comment (splincode) [inline projects/addon-doc/src/components/navigation/navigation.component.ts:170]:
```
.subscribe(() => this.navigateToAnchorLink(hash));
```

**Review by splincode — APPROVED** [review]

File: projects/addon-doc/src/components/navigation/navigation.component.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/addon-doc/src/components/navigation/navigation.component.ts]: fixed

File: projects/addon-doc/src/tokens/page-loaded.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/addon-doc/src/tokens/page-loaded.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**